protocol/src/protocol.ts

```diff
@@ -808,7 +823,7 @@ export interface _ServerCapabilities {
 	/**
 	 * The server provides rename support.
 	 */
-	renameProvider?: boolean;
+	renameProvider?: boolean | RenameOptions;
```

Comment:
We need to document here that the RenameOptions type is only valid if the client signaled prepareSupport in its capabilities. Otherwise we would break old clients.